crates/autopilot/src/domain/auction/order.rs

```diff
@@ -27,6 +28,7 @@ pub struct Order {
     pub sell_token_balance: SellTokenSource,
     pub buy_token_balance: BuyTokenDestination,
     pub app_data: AppDataHash,
+    pub full_app_data: Option<ValidatedAppData>,
```

Comment:
app data is now mandatory. Updating the `model::order` might be an unrelated change, but we should probably start expecting it to be present (or otherwise filter our orders for which we don't have this information) at least for the autopilot domain.

Reply:
@fleupold being that the case, the issue [#2505 ](https://github.com/cowprotocol/services/issues/2505) should be solved with this PR, right? because now in `protocol_fee.apply()` I can just simply assemble `ValidatedAppData` to check the `partnerFee` field.

I will clean it up and have it ready soon then.